bitswap.go

```diff
@@ -163,6 +167,7 @@ func New(parent context.Context, network bsnet.BitSwapNetwork,
 	go func() {
 		<-px.Closing() // process closes first
 		cancelFunc()
+		notif.Shutdown()
```

Comment:
Are we sure we're unsubscribing correctly? We should make sure that canceling the session context cancels all outstanding `GetBlocks` calls.

I _think_ we may need to join the session context to the `GetBlocks` context somehow (i.e., cancel the GetBlocks context when the session is closed, or something like that).

Reply:
Ah yes I think you're right, cancelling the Session context won't close the `GetBlocks()` output channel.

The `SessionManager` ties the Bitswap context to the Session context by
- creating a cancellable Session context (line 72)
- calling `cancel()` on the Session context (line 82) when the Bitswap context is done (line 86).
https://github.com/ipfs/go-bitswap/blob/a5fe0d4b62e7f6ea20264e1d1eab4d3d7e389c90/sessionmanager/sessionmanager.go#L68-L88

The `Getter` will [close the outgoing blocks channel](https://github.com/ipfs/go-bitswap/blob/master/getter/getter.go#L91) if the `GetBlocks()` context is [cancelled or times out](https://github.com/ipfs/go-bitswap/blob/master/getter/getter.go#L108).

However the `GetBlocks()` context is not tied to the Session context.